### PR TITLE
Test decode event log

### PR DESCRIPTION
**Pull request by yrong**

### CODE REVIEW

File: bridges/snowbridge/primitives/router/src/inbound/mod.rs [hunk 1]
```diff
@@ -104,7 +104,7 @@ pub struct MessageToXcm<
 }
 
 /// Reason why a message conversion failed.
-#[derive(Copy, Clone, TypeInfo, PalletError, Encode, Decode, RuntimeDebug)]
+#[derive(Copy, Clone, TypeInfo, PalletError, Encode, Decode, RuntimeDebug, PartialEq)]
```
Comment (claravanstaden) [inline bridges/snowbridge/primitives/router/src/inbound/mod.rs:107]: Why not also derive `PartialEq` when std is enabled like in https://github.com/Snowfork/polkadot-sdk/pull/119/files#diff-832917c2fd6302a9f175210f7d9941ea82a1a7c5a35aa35e662380074d405203R167?

File: cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/tests/snowbridge.rs [hunk 2]
```diff
@@ -206,3 +212,40 @@ fn construct_and_apply_extrinsic(
...
+				],
+				data: hex!("000000000000000000000000000000000000000000000000000000000000001d0000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000007300a736aa000000000001fff9976782d46cc05630d1f6ebab18b2324d6b1401ee070000904411b468d36fc59a7440fac24170a40f5c74d3a6571cf3de326300cd43874a0000000000000000000000000000000000ac23fc06000000000000000000000000e40b5402000000000000000000000000000000000000000000000000").into(),
+			};
+			let envelope =
+				snowbridge_pallet_inbound_queue::Envelope::try_from(&event_log).map_err(|_| Error::<Runtime>::InvalidEnvelope).unwrap();
+			let message = inbound::VersionedMessage::decode_all(&mut envelope.payload.as_ref()).map_err(|_| Error::<Runtime>::InvalidPayload).unwrap();
+			let (xcm,fee) = <snowbridge_pallet_inbound_queue::Pallet<Runtime>>::do_convert(envelope.message_id, message).unwrap();
+			println!("xcm converted as {:?} and fee is {:?}.", xcm, fee);
+			let result = <snowbridge_pallet_inbound_queue::Pallet<Runtime>>::send_xcm(xcm, 1000.into());
+			assert_err!(result,<Error<Runtime>>::Send(SendError::ExceedsMaxMessageSize));
```
Comment (claravanstaden) [inline cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/tests/snowbridge.rs:249]:
```suggestion
			assert_err!(result, <Error<Runtime>>::Send(SendError::ExceedsMaxMessageSize));
```

😄 nit, apologies

File: cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/tests/snowbridge.rs [hunk 3]
```diff
@@ -206,3 +212,40 @@ fn construct_and_apply_extrinsic(
 	let r = Executive::apply_extrinsic(xt);
 	r.unwrap()
 }
+
+// The event log from https://bridgehub-rococo.stg.subscan.io/extrinsic/2583529-2 to reproduce an issue
+// run this test with ```cargo test --release -p bridge-hub-rococo-runtime --test snowbridge
+// send_token_to_foreign_chain_with_invalid_dest_fee -- --nocapture```
+#[test]
+#[cfg(not(debug_assertions))]
+pub fn send_token_to_foreign_chain_with_invalid_dest_fee() {
```
Comment (claravanstaden) [inline cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/tests/snowbridge.rs:221]: Maybe I am missing the point here, but what purpose does this test serve? Is it just to replicate what happen in Rococo?

Comment (yrong) [reply cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/tests/snowbridge.rs:221]: Yeah, this PR is more of reproduce than fix. No need to merge if [1148](https://github.com/Snowfork/snowbridge/pull/1148) got accepted.

Comment (alistair-singh) [reply cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/tests/snowbridge.rs:221]: Yeah, agree. Why would a zero fee cause an error like SendError::ExceedsMaxMessageSize? Do we know the source of the error?

Comment (yrong) [reply cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/tests/snowbridge.rs:222]: Confirmed, zero Fungible asset is not allowed which fail the check [here](https://github.com/Snowfork/polkadot-sdk/blob/66505a77858d5091438f5362f53cf5d317274c30/polkadot/xcm/src/v4/asset.rs#L285)

Though the error message could be improved(`ExceedsMaxMessageSize` a bit confusing) but not the scope of this PR. So https://github.com/Snowfork/snowbridge/pull/1148 should be the right fix.

Comment (alistair-singh) [reply cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/tests/snowbridge.rs:222]: Nice find.

Do we want to consider changing the generated xcm to not BuyExecution with zero fees?
```diff
diff --git a/bridges/snowbridge/primitives/router/src/inbound/mod.rs b/bridges/snowbridge/primitives/router/src/inbound/mod.rs
index c20554c6d1..bff3b15f1a 100644
--- a/bridges/snowbridge/primitives/router/src/inbound/mod.rs
+++ b/bridges/snowbridge/primitives/router/src/inbound/mod.rs
@@ -254,16 +254,22 @@ where
 
 		match dest_para_id {
 			Some(dest_para_id) => {
-				let dest_para_fee_asset: Asset = (Location::parent(), dest_para_fee).into();
+				let mut assets = vec![asset.clone()];
+				let mut execution = UnpaidExecution { weight_limit: Unlimited, check_origin: None };
+				if dest_para_fee > 0 {
+					let dest_para_fee_asset: Asset = (Location::parent(), dest_para_fee).into();
+					assets.push(dest_para_fee_asset.clone());
+					execution = UnpaidExecution { weight_limit: Unlimited, check_origin: None };
+				}
 
 				instructions.extend(vec![
 					// Perform a deposit reserve to send to destination chain.
 					DepositReserveAsset {
-						assets: Definite(vec![dest_para_fee_asset.clone(), asset.clone()].into()),
+						assets: Definite(assets),
 						dest: Location::new(1, [Parachain(dest_para_id)]),
 						xcm: vec![
 							// Buy execution on target.
-							BuyExecution { fees: dest_para_fee_asset, weight_limit: Unlimited },
+							execution,
 							// Deposit asset to beneficiary.
 							DepositAsset { assets: Definite(asset.into()), beneficiary },
 						]

```

Comment (yrong) [reply cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/tests/snowbridge.rs:222]: I prefer to not introduce it, IMO usually parachains won't allow zero fees so more of an input error here. Even if they allow that it also requires some custom xcm config to `AllowUnpaidExecutionFrom` ony from AssetHub(not from other chain) which we not test yet.

**Comment by yrong** [comment]
Closed for https://github.com/Snowfork/polkadot-sdk/pull/119#discussion_r1510836002

**Pull request closed without merging**